client/src/commons/Forms/PlacesTypesAndSubTypes/PlacesTypesAndSubTypes.tsx

```diff
@@ -16,47 +16,39 @@ const placeSubTypeDisplayName = 'תת סוג';
 
 const PlacesTypesAndSubTypes: React.FC<Props> = (props: Props): JSX.Element => {
 
-    const { control, placeTypeName, placeSubTypeName, placeType, placeSubType,
-            onPlaceTypeChange, onPlaceSubTypeChange, required,
+    const { control, setValue, placeTypeName, placeSubTypeName, placeType, placeSubType,
+            onPlaceTypeChange, onPlaceSubTypeChange,
     } = props;
 
     const formClasses = useFormStyles();
 
     const [placesSubTypesByTypes, setPlacesSubTypesByTypes] = useState<PlacesSubTypesByTypes>({});
-    const [selectedPlaceType, setSelectedPlaceType] = useState<string>('');
-    const [selectedPlaceSubType, setSelectedPlaceSubType] = useState<PlaceSubType>();
-    const [selectedPlaceSubTypeInput, setSelectedPlaceSubTypeInput] = useState<string>('');
+    const [placeTypeInput, setPlaceTypeInput] = useState<string>('');
+    const [placeSubTypeInput, setPlaceSubTypeInput] = useState<string>('');
 
-    React.useEffect(() => {
+    useEffect(() => {
         if (Object.keys(placesSubTypesByTypes).length > 0 && placeType === '') {
             onPlaceTypeChange(Object.keys(placesSubTypesByTypes)[0]);
         }
     }, [placesSubTypesByTypes]);
 
-    React.useEffect(() => {
-        placeType && setSelectedPlaceType(placeType);
-        
+    useEffect(() => {
         if (placesSubTypesByTypes[placeType]) {
             const defaultPlaceSubType = placesSubTypesByTypes[placeType][0];
             if (defaultPlaceSubType && !placesSubTypesByTypes[placeType].map(type => type.id).includes(placeSubType)) {
-                onPlaceSubTypeChange(defaultPlaceSubType.id);
+                onPlaceSubTypeChange(defaultPlaceSubType);
             }
         }
     }, [placeType]);
 
-    React.useEffect(() => {
-        if (typeof placeSubType === 'number') {
-            setSelectedPlaceSubType(placeSubTypeById(placeSubType));
+    useEffect(() => {
+        if (placeSubTypeById(placeSubType)) {
```

Comment:
since youre using a few times maybe pass placeSubTypeById(placeSubType) to a const to avoid running through the arr several times? 🤷‍♀️